### PR TITLE
Update gradle.  Update android target&compile version to 25

### PR DESCRIPTION
**Pull request by edenman**
@JakeWharton

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,9 +16,9 @@ allprojects {
 
 ext {
   minSdkVersion = 8
-  targetSdkVersion = 23
-  compileSdkVersion = 23
-  buildToolsVersion = '23.0.0'
+  targetSdkVersion = 25
+  compileSdkVersion = 25
+  buildToolsVersion = '25.0.2'
```
Comment (JakeWharton) [inline build.gradle]: Update .travis.yml

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ language: android
 android:
   components:
     - build-tools-25.0.2
```
Comment (JakeWharton) [inline .travis.yml]: looks like you need to install `tools` and then `platform-tools` before these two

**Comment by edenman** [comment]
` Error: The SDK platform-tools version (23.0.1) is too old  to check APIs compiled with API 25; please update [NewApi]`

**Comment by edenman** [comment]
I grepped the source, we're not referencing 23.0.1 anywhere. perplexing.

**Comment by edenman** [comment]
:sob: https://github.com/travis-ci/travis-ci/issues/6040#issuecomment-219367943

**Comment by JakeWharton** [comment]
This will be fixed in 2.4.0... but we can't depend on pre-release versions.

**Comment by edenman** [comment]
ok so java8 succeeds.  java7 getting a "Unsupported major.minor version 52.0" from the LibraryPlugin...ok if I just remove the java7 build in travis.yml?

**Pull request merged**